llvm/lib/Transforms/IPO/PassManagerBuilder.cpp

```diff
@@ -369,7 +372,8 @@ void PassManagerBuilder::addPGOInstrPasses(legacy::PassManagerBase &MPM,
     MPM.add(createFunctionInliningPass(IP));
     MPM.add(createSROAPass());
     MPM.add(createEarlyCSEPass());             // Catch trivial redundancies
-    MPM.add(createCFGSimplificationPass());    // Merge & remove BBs
+    if (!SYCLOptimizationMode)
+      MPM.add(createCFGSimplificationPass());    // Merge & remove BBs
```

Comment:
I'm not sure if clang supports PGO in SYCL mode... it would be great, but it's definitely not tested regularly.